### DISCUSSION TITLE
chakra factory function doesn't work

### DISCUSSION THREAD

**Discussion by maryamseyedi** [chakra-ui/chakra-ui#5099] [Q&A]
Hello dear chakra community,

I am experiencing an issue with the Chakra Factory function. I created my DataTable component which uses and returns a chakra Table component inside. Right before exporting it I wrap it in the chakra function in a constant then I export this constant. In the parent component where I import this however, although the style props I'm using receive 0 errors, I don't see them showing up after rendered. What am I doing wrong?

Here is my code:
![image](https://user-images.githubusercontent.com/9810835/141969363-d9d4977b-75b6-487c-87db-5d0474c5515a.png)
![image](https://user-images.githubusercontent.com/9810835/141969442-706e0453-31a8-48ce-89ea-5c81c23074b1.png)
![image](https://user-images.githubusercontent.com/9810835/141969485-1bee1c6d-ac6c-434f-a41e-63c9632e28d7.png)

![image](https://user-images.githubusercontent.com/9810835/141969526-1a57ed1e-37ed-4a6f-a86a-b805f2c65075.png)

**Comment by primos63** [comment]
Why do you need to wrap `DataTable` with `chakra`?
> although the style props I'm using receive 0 errors, I don't see them showing up after rendered
What styling aren't you seeing? I have a CodeSandbox example and wrapped the `DataTable` with `chakra` and saw no difference.

**Reply by maryamseyedi** [reply]
Because if I don't I can't apply chakra's styling props after importing DataTable. I get an error `Property 'bg' does not exist on type 'IntrinsicAttributes & dataTableProps'`.

**Reply by primos63** [reply]
If you could provide a CodeSandbox example of what you are seeing, I would greatly appreciate it.

**Reply by maryamseyedi** [reply]
Here you go. I appreciate the effort. Btw I changed my method to using ChakraComponent and the exact same problem remains. 

[Link to codesandbox ](https://codesandbox.io/s/priceless-wiles-hpjto?file=/src/App.tsx)

**Reply by primos63** [reply]
Thanks! I will take a look.

**Reply by primos63** [reply]
First thing that hit me was that you didn't wrap your `DataTable` in the `ChakraProvider` in `App.tsx`
```jsx
  return (
    <ChakraProvider>
      <DataTable
        tableData={tableData}
        columns={columns}
        variant="striped"
        w={1000}
      />
    </ChakraProvider>
  );
```
When I added that the styling changed and the rows were highlighted in green on hover. Is that what you were expecting?

**Reply by maryamseyedi** [reply]
thanks for taking a look. However that wasn't the issue, as I had in my local app the provider included. the issue was with the way we pass the props. we solved it like the following:

```
const Card = (({text,...props}:{text:string,props: BoxProps}) => (

  <Box px="4" py="5" rounded="sm" shadow="lg" {...props} >{text}</Box>

)) as DivComponen
```t

**Reply by alixroyere** [reply]
Having the same issue, according to the current doc:

> Considering that Chakra uses emotion under the hood, ensure the non-chakra component accepts className as props for this to work correctly

https://chakra-ui.com/docs/styled-system/chakra-factory

I tested that and having only `className` in input props of custom component and passing it down in the component works properly

```
const Card = (props:{text: string, className?: string}) => (

  <Box className={props.className} >{props.text}</Box>

) 
```
